### PR TITLE
IC-1627 clarify the behaviour around ending referrals prematurely vs concluding by submitting an EOSR

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

please see the individual commits for details on the contents of this PR.

*GROUP REVIEW REQUIRED*

https://dsdmoj.atlassian.net/browse/IC-1627

https://docs.google.com/spreadsheets/d/1Jupmyv4nFpKv96G3XDyjkCLYjjwRRQP2pH-rRxgyUDY/edit#gid=479692448

## What is the intent behind these changes?

clarify the behaviour around ending referrals prematurely vs concluding by submitting an EOSR

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/integration/pact/SetupAssistant.kt [hunk 1]
```diff
@@ -144,7 +144,7 @@ class SetupAssistant(
   fun createEndedReferral(id: UUID = UUID.randomUUID(), intervention: Intervention = createIntervention(), cancellationReason: CancellationReason? = randomCancellationReason(), cancellationComments: String? = null): Referral {
```
Comment (rgforsyth) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/integration/pact/SetupAssistant.kt:144]: We should probably change cancellationReason/comments etc in the method parameters to the new names.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralServiceUnitTest.kt [hunk 2]
```diff
@@ -44,13 +44,13 @@ class ReferralServiceUnitTest {
     val cancellationComments = "comment"
```
Comment (rgforsyth) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralServiceUnitTest.kt:49]: As above

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/util/ReferralFactory.kt [hunk 3]
```diff
@@ -131,10 +131,10 @@ class ReferralFactory(em: TestEntityManager? = null) : EntityFactory(em) {
     assignedAt: OffsetDateTime? = null,
     assignedBy: AuthUser? = null,
     assignedTo: AuthUser? = null,
-    endedAt: OffsetDateTime? = null,
-    endedBy: AuthUser? = null,
-    cancellationReason: CancellationReason? = null,
-    cancellationComments: String? = null,
+    endRequestedAt: OffsetDateTime? = null,
+    endRequestedBy: AuthUser? = null,
+    endRequestedReason: CancellationReason? = null,
```
Comment (rgforsyth) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/util/ReferralFactory.kt:142]: CancellationReason class could also be renamed?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/integration/pact/SetupAssistant.kt (hunk 1 above)
Comment (tomsmyers) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/integration/pact/SetupAssistant.kt]: done

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralServiceUnitTest.kt (hunk 2 above)
Comment (tomsmyers) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralServiceUnitTest.kt:49]: changed the params, left the vars for now

**Review by rgforsyth — APPROVED** [review]
👍

**Pull request merged**